examples/gno.land/p/demo/uassert/uassert.gno

```diff
@@ -416,3 +419,317 @@ func NotEmpty(t TestingT, obj interface{}, msgs ...string) bool {
 	}
 	return true
 }
+
+// Nil asserts that the specified object is nil.
+func Nil(t TestingT, obj interface{}, msgs ...string) bool {
+	t.Helper()
+	if obj != nil {
+		return fail(t, msgs, "uassert.Nil: object should be nil")
+	}
+
+	return true
+}
+
+// NotNil asserts that the specified object is not nil.
+func NotNil(t TestingT, obj interface{}, msgs ...string) bool {
+	t.Helper()
+	if obj == nil {
+		return fail(t, msgs, "uassert.NotNil: object should not be nil")
+	}
+
+	return true
+}
+
+// Contains asserts that the specified string or list contains the specified substring or element.
+func Contains(t TestingT, container, element interface{}, msgs ...string) bool {
+	t.Helper()
+	switch c := container.(type) {
+	case string:
+		if e, ok := element.(string); ok {
+			if !strings.Contains(c, e) {
+				return fail(t, msgs, "uassert.Contains: %q does not contain %q", c, e)
+			}
+			return true
+		}
+	case []string:
+		for _, item := range c {
+			if item == element {
+				return true
+			}
+		}
+		return fail(t, msgs, "uassert.Contains: %v does not contain %v", container, element)
+	default:
+		return fail(t, msgs, "uassert.Contains: unsupported type %T", container)
+	}
+
+	return false
+}
+
+// NotContains asserts that the specified string or list does not contain the specified substring or element.
+func NotContains(t TestingT, container, element interface{}, msgs ...string) bool {
+	t.Helper()
+	switch c := container.(type) {
+	case string:
+		if e, ok := element.(string); ok {
+			if strings.Contains(c, e) {
+				return fail(t, msgs, "uassert.NotContains: %q contains %q", c, e)
+			}
+			return true
+		}
+	case []string:
+		for _, item := range c {
+			if item == element {
+				return fail(t, msgs, "uassert.NotContains: %v contains %v", container, element)
+			}
+		}
+		return true
+	default:
+		return fail(t, msgs, "uassert.NotContains: unsupported type %T", container)
+	}
+	return false
+}
+
+// Len asserts that the specified object has specific length.
+func Len(t TestingT, obj interface{}, length int, msgs ...string) bool {
+	t.Helper()
+	actualLen := -1
+	switch v := obj.(type) {
+	case string:
+		actualLen = len(v)
+	case []interface{}:
+		actualLen = len(v)
```

Comment:
This will only work if obj is exactly an `[]interface{}`; ie., not if is a `[]int` or anything else.

Personally, I would prefer `uassert.Equal(t, 100, len(p), "p should have 100 items")`

Reply:
I've added more object types supported by the `Len` function. I agree with your use of `len(p)` directly. But this function is part of the original testify package, and I thought it would be nice to add it anyway, so that it's available if other people prefer to use this function.

https://github.com/gnolang/gno/pull/2547/commits/b99e866001a1fbe240503f1fc3994504241bb4a1